git_hooks/pre-commit

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# To use this hook, you can configure this directory as the git `hooksPath`:
+#
+#   git config core.hooksPath git_hooks
+#
+
+set -euo pipefail
+IFS=$'\n\t '
+
+lint_failed=false
+
+run_lint() {
+  if [ ! -f ./node_modules/.bin/${LINTER} ];
+  then
+    echo "linter not installed - please run \"npm install\" first"
+  fi
+
+  # Only lint files that were added, copied, modified, or renamed
+  for file in $(git diff --diff-filter=ACMR --cached --name-only . | grep -E ${LINTER_PATTERN}); do
+    local error=0
+
+    git show ":$file" | \
+      ./node_modules/.bin/${LINTER} ${LINTER_ARGS[@]} || error=1
```

Comment:
I don't quite understand what this line does. Specifically I don't understand what `git show` does when passed a file.

Reply:
It shows the staged file. This way we check what is actually about to be commited, which can be different from what is currently on disk.

Reply:
Cool. Merge away